clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -156,6 +397,112 @@ else if (key.defaultValue == NO_DEFAULT_VALUE)
         return values;
     }
 
+
+    /**
+     * Validate the provided connector configuration values with the configuration definition
+     * @param connectorConfigs the provided connector configuration values
+     * @return List of Config, each Config contains the updated configuration information given
+     * the current configuration values.
+     */
+    public List<ConfigValue> validate(Map<String, String> connectorConfigs) {
+        List<String> configsWithNoParent = getConfigsWithNoParent();
+        Map<String, ConfigValue> configValues = new HashMap<>();
+
+        for (String name: configKeys.keySet()) {
+            configValues.put(name, new ConfigValue(name));
+        }
+
+        for (String name: configsWithNoParent) {
+            validate(name, connectorConfigs, configValues);
+        }
+        return new LinkedList<>(configValues.values());
+    }
+
+    private void ensureDependentsDefined() {
+        for (String configName: configKeys.keySet()) {
+            ConfigKey configKey = configKeys.get(configName);
+            List<String> dependents = configKey.dependents;
+            for (String dependent: dependents) {
+                if (!configKeys.containsKey(dependent)) {
+                    throw new ConfigException(dependent + " is  an dependent of " + configName + ". However, it is not defined.");
+                }
+            }
+        }
+    }
+
+    private List<String> getConfigsWithNoParent() {
+        Set<String> configs = configKeys.keySet();
+        Set<String> configsWithParent = new HashSet<>();
+
+        for (ConfigKey configKey: configKeys.values()) {
+            List<String> dependents = configKey.dependents;
+            configsWithParent.addAll(dependents);
+        }
+
+        List<String> configsWithNoParent = new LinkedList<>();
+        for(String config: configs) {
+            if (!configsWithParent.contains(config)) {
+                configsWithNoParent.add(config);
+            }
+        }
+        return configsWithNoParent;
+    }
+
+    private void validate(String name, Map<String, String> connectorConfigs, Map<String, ConfigValue> configs) {
```

Comment:
Yeah, a different (more general) name. Despite the fact that this is only used by connectors at the moment, this class is part of the common client utilities. "Connect" names generally shouldn't leak into this layer.